model/task_lifecycle.go

```diff
@@ -31,109 +31,114 @@ type StatusChanges struct {
+								return errors.Wrapf(err, "error resetting dependency '%s'", dep.Id)
+							}
+						} else {
+							tasksToActivate = append(tasksToActivate, dep)
 						}
-					} else {
-						tasksToActivate = append(tasksToActivate, dep)
 					}
+				} else {
+					tasksToActivate = append(tasksToActivate, deps...)
 				}
-			} else {
-				tasksToActivate = append(tasksToActivate, deps...)
 			}
-		}
 
-		// Investigating strange dispatch state as part of EVG-13144
-		if !utility.IsZeroTime(t.DispatchTime) && t.Status == evergreen.TaskUndispatched {
-			if err := resetTask(t.Id, caller, false); err != nil {
-				return errors.Wrap(err, "error resetting task")
+			// Investigating strange dispatch state as part of EVG-13144
+			if !utility.IsZeroTime(t.DispatchTime) && t.Status == evergreen.TaskUndispatched {
+				if err := resetTask(t.Id, caller, false); err != nil {
+					return errors.Wrap(err, "error resetting task")
+				}
+			} else {
+				tasksToActivate = append(tasksToActivate, originalTasks...)
 			}
-		} else {
-			tasksToActivate = append(tasksToActivate, originalTasks...)
-		}
-		if err := task.ActivateTasks(tasksToActivate, time.Now(), caller); err != nil {
-			return errors.Wrapf(err, "can't activate tasks")
-		}
-
-		if t.DistroId == "" && !t.DisplayOnly {
-			grip.Critical(message.Fields{
-				"message": "task is missing distro id",
-				"task_id": t.Id,
-			})
-		}
 
-		version, err := VersionFindOneId(t.Version)
-		if err != nil {
-			return errors.Wrapf(err, "error find associated version")
-		}
-		if version == nil {
-			return errors.Errorf("could not find associated version : `%s`", t.Version)
-		}
-		if err = version.SetActivated(); err != nil {
-			return errors.Wrapf(err, "Error marking version as activated")
-		}
+			if t.DistroId == "" && !t.DisplayOnly {
+				grip.Critical(message.Fields{
+					"message": "task is missing distro id",
+					"task_id": t.Id,
+				})
+			}
 
-		// If the task was not activated by step back, and either the caller is not evergreen
-		// or the task was originally activated by evergreen, deactivate the task
-	} else if !evergreen.IsSystemActivator(caller) || evergreen.IsSystemActivator(t.ActivatedBy) {
-		var err error
-		// deactivate later tasks in the group as well, since they won't succeed without this one
-		if t.IsPartOfSingleHostTaskGroup() {
-			tasksInGroup, err := task.FindTaskGroupFromBuild(t.BuildId, t.TaskGroup)
+			version, err := VersionFindOneId(t.Version)
 			if err != nil {
-				return errors.Wrapf(err, "error finding task group '%s'", t.TaskGroup)
+				return errors.Wrapf(err, "error find associated version")
+			}
+			if version == nil {
+				return errors.Errorf("could not find associated version : `%s`", t.Version)
 			}
-			for _, taskInGroup := range tasksInGroup {
-				if taskInGroup.TaskGroupOrder > t.TaskGroupOrder {
-					originalTasks = append(originalTasks, taskInGroup)
+			versionsToActivate = append(versionsToActivate, *version)
```

Comment:
Let's just keep a list of versionIdsToActivate so that we don't have to keep re-querying the version (especially since these will all commonly be for the same version) -- we can just use this list directly then in ActivateVersions